src/frontend/js/modules/session.spec.js

```diff
@@ -366,4 +366,66 @@ describe('The esn.session Angular module', function() {
 
   });
 
+  describe('the sessionInitESNController', () => {
+    let userAPI, esnAuth, self;
+
+    beforeEach(function() {
+      self = this;
+
+      userAPI = {
+        currentUser: sinon.stub().returns($q.when({})),
+        user: sinon.stub().returns($q.when({}))
+      };
+
+      esnAuth = {
+        signInCompletePromise: {
+          then: callback => callback()
+        },
+        signout: sinon.spy(),
+        init: sinon.stub().returns($q.when({}))
+      };
+
+      angular.mock.module(function($provide) {
+        $provide.value('userAPI', userAPI);
+        $provide.value('esnAuth', esnAuth);
+      });
+
+      angular.mock.inject(function($rootScope, $controller, session) {
+        this.$scope = $rootScope.$new();
+        this.$rootScope = $rootScope;
+        this.$controller = $controller;
+        this.session = session;
+      });
+    });
+
+    function initController() {
+      const controller = self.$controller('sessionInitESNController', {
+        $rootScope: self.$rootScope,
+        $scope: self.$scope
+      });
+
+      return controller;
+    }
+
+    it('should signout the user if it is unauthorized ( broken session )', function() {
+      esnAuth.init = sinon.stub().returns($q.reject({
```

Comment:
You don't test exactly what are you doing. The failed promise you should add test is: `sessionFactory.bootstrapSession()`

Reply:
the error happens when the init function returns a 401 error instead of resolving the user

Reply:
i'll update the logger message so it will be more clear